src/frontend/js/hooks/useTeacherCoursesSearch/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { useCourseProductUnion } from 'hooks/useCourseProductUnion';
+import { CourseListItem, CourseProductRelation, ProductType } from 'types/Joanie';
+
+const useTeacherCoursesSearch = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [courseAndProductList, setCourseAndProductList] = useState<
+    (CourseListItem | CourseProductRelation)[]
+  >([]);
+  const [isInitialLoading, setIsInitialLoading] = useState(false);
+
+  const { data, isLoading, next, hasMore, count } = useCourseProductUnion({
+    query: searchParams.get('query') || undefined,
+    perPage: 25,
+    productType: ProductType.CREDENTIAL,
+  });
+
+  useEffect(() => {
+    if (!data.length && isLoading) {
```

Comment:
what if `data.length` is 0 ?

Reply:
That the point, when `data.length` is 0 with `isLoading === true` then it mean that we've trigger a new research.